### ISSUE TITLE
Alphazero/Kramnik variants

### ISSUE THREAD

**Issue by GregNeto** [ianfab/Fairy-Stockfish#209]
https://en.chessbase.com/post/alphazero-kramnik-exploring-new-chess-variants

I especially like torpedo chess, great rule simpification, fun and different gameplay.

I have posted some perft results but nobody seemed to care.

http://talkchess.com/forum3/viewtopic.php?t=75114

Tx

**Comment by ianfab** [comment]
Thanks for your suggestion. 

I just copy&paste the table from the article with short comments by me:

Variant | Primary rule change | Secondary rule change | comment
-- | -- | -- | --
No-castling | Castling is disallowed throughout the game | -- | available
No-castling (10) | Castling is disallowed for the first 10 moves (20 plies) | -- | IMO not worth implementing
Pawn one square | Pawns can only move by one square | -- | available
Stalemate=win | Forcing stalemate is a win rather than a draw | -- | available
Torpedo | Pawns can move by 1 or 2 squaresanywhere on the board. En passant can happen anywhere on the board. | -- | might be doable, unsure
Semi-torpedo | Pawns can move by two square both from the 2nd and the 3rd rank | -- | probably doable, similar to horde
Pawn-back | Pawns can move backwards by one square, but only back to the 2nd/7th rank for White/Black | Pawn moves do not count towards the 50 move rule | alternative pawns are not supported yet
Pawn-sideways | Pawns can also move laterally by one square. Captures are unchanged, diagonally upwards | Sideway pawn moves do not count towards the 50 move rule | alternative pawns are not supported yet
Self-capture | It is possible to capture one’s own pieces | -- | probably difficult

**Comment by mtaktikos** [comment]
In the category "alternative pawns" is also issue #51 (Berolina chess)

**Comment by ianfab** [comment]
Yes, although berolina pawns are a bit more problematic, because they do not just add but alter the movements and captures, and there also is the altered en passant. I am still not clear about how to implement them in a clean way.

I merged support for semi-torpedo to master, since that was a very natural extension of the first rank double steps in horde. The other variants would require to implement new pawn movements (torpedo needs promotion with double steps), and since the respective variants are the only ones that use their special movements, I am currently a bit sceptical if they are worth the overhead for performance and maintenance.

**Comment by musketeerchess** [comment]
Hi
Pawns are described as the soul of chess by Morphy. It's for a good reason,
messing with them alters the game in numerous ways and this is not always
for the good.

The best alternative right now for the Classic Pawn is the Berolina Pawn.
The game becomes much more tactical and the endgames are usually won by one
side or the other. This is because it's difficult to have "walls" and
closed positions and double pawns are very strong !!

Other interesting alternate pawns are those with lateral moves (helping
avoid walls) or simply the possibility to play a backwards move which also
can be a good defense but also avoids closing of the position.

I made a few years ago some trials to extend the pawns rules.
My favorite is Lieutenant featured in Moscovia Chess.

It can move one square backwards (not possible on initial square naturally)
it can also move 1 square laterally (and these additional moves are non
capturing moves) when on initial square or on the 7th row.

You can check the rules and play a game here
https://musketeerchess.net/games/sp/lieutenant/

Other interesting alternate pawns are the :
General https://musketeerchess.net/games/sp/general/rules/rules.php
Colonel https://musketeerchess.net/games/sp/colonel/rules/rules.php
Commander https://musketeerchess.net/games/sp/commander/rules/rules.php
Captain https://musketeerchess.net/games/sp/captain/rules/rules.php
Sergeant https://musketeerchess.net/games/sp/sergeant/rules/rules.php
Corporal  https://musketeerchess.net/games/sp/corporal/rules/rules.php






Le sam. 21 nov. 2020 à 23:37, Fabian Fichter <notifications@github.com> a
écrit :
